packages/core/core/src/Transformation.js

```diff
@@ -388,7 +394,11 @@ export default class Transformation {
   }
 
   async readFromCache(cacheKey: string): Promise<?Array<UncommittedAsset>> {
-    if (this.options.shouldDisableCache || this.request.code != null) {
+    if (
+      this.options.shouldDisableCache ||
+      this.request.code != null ||
+      (this.request.invalidateReason || 0) & FILE_CREATE
```

Comment:
File create invalidations aren't in the cache key (what would we put in it?), so without this the transformation would never invalidate when a new file was created.

Reply:
I was thinking of the case where the asset wasn't in the request graph but there's a transformation cache entry. But I think that doesn't hit the cache at the moment anyway because the invalidations aren't known yet

Reply:
Yeah that's certainly an issue, but not really sure how to solve it 🤔